### DISCUSSION TITLE
Choice on which network to connect to on init

### DISCUSSION THREAD

**Discussion by MartinGerritsen** [family/connectkit#64] [Ideas]
Hi, great work here and I'm a happy user! I am now working on a website which works on 2 chains though, Matic and a custom new chain (Alt) and the users will be quite divided between the two - but because of my chain order it always automatically connects to Matic after which we can open the modal and change the network again, quite a few extra steps especially since the connection resets on refresh.

I know of the initialChainId that I can set, but this is less than desired and feels like a hack to check if a wallet is already open in the browser and possibly take the connected chain id and set it in the ConnectKitProvider on init ... 

Could an option be added to, when more than one chain has been given to Wagmi, we first get a Chain selector modal before trying to connect? 

Also, is there a way to customize the modal in such a big way that changing networks doesn't need a dropdown to be opened etc? 

Thank you!

**Comment by dirtycajunrice** [comment]
From another user's perspective:
UI design of the network selection is 3 clicks total. Wallet button > network dropdown > network.
This really only can be reduced by 1 click and im not sure its worth the effort. It is a pretty design as-is.
That said, im sure some users would see benefit from a "modal list" instead of a dropdown list. Just a matter of development priority i guess... as there are things that are missing from core implementation that probably have higher priority and you can use wagmi to make the same modal yourself.

As for the "desiredNetworkID", my suggestion is for you to either save the user's preferred network in local storage and grab that prior to initializing, or query the wallet for the chainId via window.ethereum prior to wagmi init, as you cant use the package to query when its the data you need to initialize (race condition)

**Reply by MartinGerritsen** [reply]
Thank you for taking time to think along and respond!

In my opinion 2 clicks (wallet click can not be avoided) is 1 too many ;) 
I might take a shot at it myself, because it is quite the bother :) Not criticism to the whole project, it's real good!
And of course I understand priority orders :) 

I do actually think that the Provider can pick up if the wallet is connected.
- if no, then we just try to connect to the configuration order on connect. (I don't like making cookies)
- if yes, is it one that is in the networks list? 
- - if no, see step 1
- - if yes, put it to the front of the array

Again, might have a hack at this myself, all that is missing is the free time!
Not saying it's easy, its interesting! :)

**Comment by lochie** [comment]
We've released an update in [ConnectKit `1.1.2`](https://github.com/family/connectkit/releases/tag/v1.1.2) where if you set the [`initialChainId` to `0`](https://docs.family.co/connectkit/api-reference#ConnectKitProvider-initialChainId) it will no longer target a chain and let the wallet connect to whichever chain it was previously set to.

Hopefully this helps resolve some issues 🙏

**Reply by MartinGerritsen** [reply]
It works like a charm, thank you!
This detail when connected to another network really finishes it as well :)
![image](https://user-images.githubusercontent.com/861346/214318494-8d05cfa0-187c-4460-af72-eae189e201e2.png)
